include/httpp/http/Connection.hpp

```diff
@@ -78,6 +78,56 @@ class Connection
         return request_;
     }
 
+
+    template <typename Callable>
+    void read(size_t size, char* buffer, Callable&& callable)
+    {
+        if (!own())
+        {
+            throw std::logic_error("Invalid connection state");
+        }
+
+        size_t offset = 0;
+
+        if (not body_buffer_.empty())
+        {
+            if (size <= body_buffer_.size())
+            {
+                std::copy(begin(body_buffer_), begin(body_buffer_) + size,
+                          buffer);
+                body_buffer_.erase(begin(body_buffer_),
```

Comment:
Do you need to erase this part of body_buffer_ in this case?  or does the connection and buffer stay valid after calling disown and the callback?